ddtrace/tracer/spancontext.go

```diff
@@ -236,6 +236,9 @@ func (t *trace) finishedOne(s *span) {
 		return
 	}
 	if tr, ok := internal.GetGlobalTracer().(*tracer); ok {
+		if s == t.root && tr.hostname != "" {
```

Comment:
There is no reason to have this here. It just confuses readers. Plus, this won't have the expected behavior in one case: the root might finish, but the trace won't (due to a child taking longer than the root or an async child).

This logic needs to be moved to (base) line  228. We should end up with something like this:
```go
    if s == t.root {
        // the root has finished; we should apply all trace-level metadata now.
	// we won't be able to make changes to a span after it finished without
	// causing a race condition.
        if t.priority != nil {
		// lock down the priority
		t.root.Metrics[keySamplingPriority] = *t.priority
		t.locked = true
        }
        if tr.hostname != "" {
            t.root.Meta[keyHostname] = tr.hostname
        }
    }
```

Reply:
It seems like you won't have access to the tracer instance here. This is getting more complicated than needed. Can we not simply add the hostname tag when we start the span? Just like we add the PID [here](https://github.com/DataDog/dd-trace-go/blob/v1/ddtrace/tracer/tracer.go#L275).